specs/spec_reader_test.go

```diff
@@ -93,3 +94,51 @@ func TestLoadSpecs(t *testing.T) {
 		})
 	}
 }
+
+func TestExpandFile(t *testing.T) {
+	cfg := []byte(`
+kind: source
+spec:
+	name: test
+	version: v1.0.0
+	spec:
+		credentials: ${file:./testdata/creds.txt}
+		otherstuff: 2
+		credentials1: ${file:./testdata/creds1.txt}
+	`)
+	expectedCfg := []byte(`
+kind: source
+spec:
+	name: test
+	version: v1.0.0
+	spec:
+		credentials: mytestcreds
+		otherstuff: 2
+		credentials1: anothercredtest
+	`)
+	expandedCfg, err := expandFileConfig(cfg)
+	if err != nil {
+		t.Fatal(err)
+	}
+	if !bytes.Equal(expandedCfg, expectedCfg) {
+		t.Fatalf("got: %s expected: %s", expandedCfg, expectedCfg)
+	}
+
+	badCfg := []byte(`
+kind: source
+spec:
+	name: test
+	version: v1.0.0
+	spec:
+		credentials: ${file:./testdata/creds2.txt}
+		otherstuff: 2
+	`)
+	_, err = expandFileConfig(badCfg)
+	expectedErr := `open ./testdata/creds2.txt: no such file or directory`
```

Comment:
Why not use `errors.Is` and ErrNotExist etc.

Reply:
yeah that would be better. I basically copy pasted from other tests :)